lib/CustomerBalanceTransaction.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Stripe;
+
+/**
+ * Class CustomerBalanceTransaction
+ *
+ * @package Stripe
+ *
+ * @property string $id
+ * @property string $object
+ * @property int $amount
+ * @property int $created
+ * @property string $currency
+ * @property string $customer
+ * @property string $description
+ * @property int $ending_balance
+ * @property string $invoice
+ * @property bool $livemode
+ * @property StripeObject $metadata
+ * @property string $type
+ */
+class CustomerBalanceTransaction extends ApiResource
+{
+    const OBJECT_NAME = "customer_balance_transaction";
+
+    /**
+     * Possible string representations of a customer balance transaction's type.
+     * @link https://stripe.com/docs/api/customers/customer_balance_transaction_object#customer_balance_transaction_object-type
+     */
+    const TYPE_ADJUSTEMENT                = 'adjustment';
+    const TYPE_APPLIED_TO_INVOICE         = 'applied_to_invoice';
+    const TYPE_CREDIT_NOTE                = 'credit_note';
+    const TYPE_INITIAL                    = 'initial';
+    const TYPE_INVOICE_TOO_LARGE          = 'invoice_too_large';
+    const TYPE_INVOICE_TOO_SMALL          = 'invoice_too_small';
+    const TYPE_MIGRATION                  = 'migration';
+    const TYPE_RECEIVER_SOURCE_MISPAYMENT = 'receiver_source_mispayment';
+    const TYPE_REVERSAL                   = 'reversal';
+
+    /**
+     * @return string The API URL for this customer balance transaction.
+     */
+    public function instanceUrl()
+    {
+        $id = $this['id'];
+        $customer = $this['customer'];
+        if (!$id) {
+            throw new Error\InvalidRequest(
+                "Could not determine which URL to request: class instance has invalid ID: $id",
+                null
+            );
+        }
+        $id = Util\Util::utf8($id);
+        $customer = Util\Util::utf8($customer);
+
+        $base = Customer::classUrl();
+        $customerExtn = urlencode($customer);
+        $extn = urlencode($id);
+        return "$base/$customerExtn/customer_balance_transactions/$extn";
```

Comment:
this all method bothers me every time I copy it. I really think there needs to be a cleaner helper that takes the name of the parent property or something and does the rest automatically.

Reply:
Agreed. In this case I think we don't even need the method, since there are no instance methods.

Reply:
Should I remove it?

Reply:
🤷‍♂ Let's keep it for now, it doesn't hurt to keep it.